classes/affiliation/DAO.php

```diff
@@ -0,0 +1,175 @@
+
+    /** @copydoc EntityDAO::$schema */
+    public $schema = PKPSchemaService::SCHEMA_AFFILIATION;
+
+    /** @copydoc EntityDAO::$table */
+    public $table = 'author_affiliations';
+
+    /** @copydoc EntityDAO::$settingsTable */
+    public $settingsTable = 'author_affiliation_settings';
+
+    /** @copydoc EntityDAO::$primaryKeyColumn */
+    public $primaryKeyColumn = 'author_affiliation_id';
+
+    /** @copydoc EntityDAO::$primaryTableColumns */
+    public $primaryTableColumns = [
+        'id' => 'author_affiliation_id',
+        'authorId' => 'author_id',
+        'ror' => 'ror'
+    ];
+
+    /**
+     * Get the parent object ID column name
+     */
+    public function getParentColumn(): string
+    {
+        return 'author_id';
+    }
+
+    /**
+     * Instantiate a new DataObject
+     */
+    public function newDataObject(): Affiliation
+    {
+        return App::make(Affiliation::class);
+    }
+
+    /**
+     * Get the number of Affiliation's matching the configured query
+     */
+    public function getCount(Collector $query): int
+    {
+        return $query
+            ->getQueryBuilder()
+            ->getCountForPagination();
+    }
+
+    /**
+     * Get a list of ids matching the configured query
+     */
+    public function getIds(Collector $query): Collection
+    {
+        return $query
+            ->getQueryBuilder()
+            ->select('a.' . $this->primaryKeyColumn)
+            ->pluck('a.' . $this->primaryKeyColumn);
+    }
+
+    /**
+     * Get a collection of affiliations matching the configured query.
+     * If the submission locale is not found in rors / ror_settings table,
+     * use the display locale name.
+     */
+    public function getMany(Collector $query, ?string $submissionLocale = null): LazyCollection
+    {
+        $rows = $query
+            ->getQueryBuilder()
+            ->get();
+
+        return LazyCollection::make(function () use ($rows, $submissionLocale) {
+            $rorIds = [];
+            foreach ($rows as $row) {
+                if ($row->ror) $rorIds[] = $row->ror;
+            }
+            $rors = iterator_to_array(Repo::ror()->getCollector()->filterByRors($rorIds)->getManyRorAsCollectionId());
+
+            foreach ($rows as $row) {
+                $fromRow = $this->fromRow($row);
+                if ($fromRow->_data['ror']) {
+                    $fromRow->_data['name'] = $rors[$fromRow->_data['ror']]->_data['name'];
+                    unset($fromRow->_data['name'][Ror::NO_LANG_CODE]);
+                    if(empty($fromRow->_data['name'][$submissionLocale])) {
+                        $displayLocale = $rors[$fromRow->_data['ror']]->_data['displayLocale'];
+                        $fromRow->_data['name'][$submissionLocale] =
+                            $rors[$fromRow->_data['ror']]->_data['name'][$displayLocale];
+                    }
+                }
+                yield $row->author_affiliation_id => $fromRow;
+            }
+        });
```

Comment:
This is not easy to read and understand, so maybe to do it similarly to author DAO and set affiliations?
To get the affiliation fromRow() and then get the Ror by ID and set the affiliations names.

Reply:
Then we would also not need that function getManyRorAsCollectionId() in Ror Collector